spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AwsSecretsManagerEnvironmentRepository.java

```diff
@@ -112,24 +112,26 @@ else if (StringUtils.isEmpty(label)) {
 
 		List<String> reversedProfiles = new ArrayList<>(Arrays.asList(profiles));
 		Collections.reverse(reversedProfiles);
+		if (!reversedProfiles.contains(defaultProfile)) {
+			reversedProfiles.add(defaultProfile);
+		}
+
+		List<String> applications = Arrays.asList(StringUtils.commaDelimitedListToStringArray(application));
+		Collections.reverse(applications);
+		if (!applications.contains(defaultApplication)) {
+			applications = new ArrayList<>(applications);
+			applications.add(defaultApplication);
+		}
+
 		for (String l : labels) {
 			for (String profile : reversedProfiles) {
-				addPropertySource(environment, application, profile, l);
-				if (!defaultApplication.equals(application)) {
-					addPropertySource(environment, defaultApplication, profile, l);
+				for (String app : applications) {
+					addPropertySource(environment, app, profile, l);
 				}
 			}
-			if (!Arrays.asList(profiles).contains(defaultProfile)) {
-				addPropertySource(environment, application, defaultProfile, l);
```

Comment:
Are there existing tests that verify that changing this logic doesn't break anything?  I am nervous that we might be changing the order of certain property sources and breaking expectations.

Reply:
[This one](https://github.com/spring-cloud/spring-cloud-config/blob/f38589ba69c28205d405d2f3bc0f20dd6ac9a600/spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/AwsSecretsManagerEnvironmentRepositoryTests.java#L781C1-L825C3) appears to provide good coverage for the logic changes in AWS Secrets Manager, particularly regarding the ordering and inclusion of profiles.

It ensure that:
- Profiles specified as a comma-separated list (like `"prod,east"`) are handled in the correct order.
- `null` profiles are properly included.
- The default profile is correctly added, and its position relative to other profiles (including `null` profiles) is maintained.

Based on my review, this test, along with `testFindOneWithNullApplicationAndNonExistingProfileAndNullLabelWhenDefaultLabelIsSet` and `testFindOneWithExistingApplicationAndNonExistingProfileAndNoDefaultProfileForFooAndNullLabelWhenDefaultLabelIsSet`, seems to cover the existing logic quite well.